### PR TITLE
Zuora: Add connector

### LINKED ISSUE #226: Add Zuora connector

### PR DESCRIPTION
**Pull request by Beadko**
## Checklist
- [x] Ran Linter
- [x] Catalog tests passing
- [x] Created PR from non-main branch 

## Catalog variables
no variables

## Notes
Resolves #226 

AuthURL used is a mock URL, since Zuora does not require a code for receiving a token.

Use /v1/ instead of /v2/ in the API calls.

### GET 
URL: <localhost:4444/v1/object/product/8a8aa2028ef98921018eff24f27d799b> 
Retrieve a Product
Screenshot 2024-04-21 at 13 33 57

### POST
URL: <localhost:4444/v1/object/product> 
Create a Product
Screenshot 2024-04-21 at 13 30 16

### PUT
URL: <localhost:4444/v1/object/product/8a8aa2028ef98921018eff24f27d799b> 
Update a Product
Screenshot 2024-04-21 at 13 36 00

### DELETE
URL: <localhost:4444/v1/object/product/8a8aa2028ef98921018eff24f27d799b> 
Delete a Product
Screenshot 2024-04-21 at 13 36 20

## Pagination
Get a list of products without pagination
Screenshot 2024-04-21 at 13 38 33

Get a list of products with pagination (3 items per page, land on page 3)
Screenshot 2024-04-21 at 13 39 29

### CODE REVIEW

File: providers/catalog.go [hunk 1]
```diff
@@ -1215,4 +1216,29 @@ var catalog = CatalogType{ // nolint:gochecknoglobals
 			Write:     false,
 		},
 	},
+
+	// Zuora Configuration
+	Zuora: {
+		AuthType: Oauth2,
+		BaseURL:  "https://rest.test.zuora.com",
```
Comment (eberle1080) [inline providers/catalog.go:1223]: I'm guessing this came from here:

https://developer.zuora.com/api-references/api/overview/#section/Base-URL

I know that this ticket is about getting the sandbox working. So in that sense this is fine. But we also need to support the production URLs. To help with that, I'd suggest changing this from a hard-coded domain to something we can substitute.

```suggestion
		BaseURL:  "https://{{.subdomain}}.zuora.com",
```

**Review by eberle1080 — CHANGES_REQUESTED** [review]

File: providers/catalog.go (hunk 1 above)
Comment (Beadko) [reply providers/catalog.go]: updated

File: providers/catalog.go [hunk 2]
```diff
@@ -1215,4 +1216,29 @@ var catalog = CatalogType{ // nolint:gochecknoglobals
...
 	},
+
+	// Zuora Configuration
+	Zuora: {
+		AuthType: Oauth2,
+		BaseURL:  "https://{{.subdomain}}.zuora.com",
+		OauthOpts: OauthOpts{
+			GrantType:                 ClientCredentials,
+			AuthURL:                   "https://rest.test.zuora.com/oauth/auth_mock",
+			TokenURL:                  "https://rest.test.zuora.com/oauth/token",
```
Comment (eberle1080) [inline providers/catalog.go:1227]: I think that these need the substitution as well

Comment (Beadko) [reply providers/catalog.go]: updated the URL's

**Review by eberle1080 — APPROVED** [review]

**Comment by eberle1080** [comment]
looks good, be sure to resolve the conflicts before you merge

**Comment by Beadko** [comment]
@eberle1080 I have rebased the branch. Don't think I have access rights to merge the PR.

**Comment by eberle1080** [comment]
oh ok, I do, I'll merge this then. Thanks

**Pull request merged**